File: sort-mapper-example-java/src/main/java/kh/farrukh/sortmapperexamplejava/course/model/CourseCreateRequestDTO.java
package kh.farrukh.sortmapperexamplejava.course.model;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Positive;
import jakarta.validation.constraints.PositiveOrZero;
import jakarta.validation.constraints.Size;

public record CourseCreateRequestDTO(
    @NotBlank
    @Size(min = 2, max = 50)
    String title,
    @NotBlank
    @Size(min = 2, max = 200)
    String description,
    @Size(min = 2, max = 100)
    String slogan,
    @NotNull
    Long teacherId,
    @NotNull
    @Positive
    Integer durationMonth,
    @NotNull
    @PositiveOrZero
    Double price
) {

}

File: sort-mapper-example/src/main/kotlin/kh/farrukh/sortmapperexample/course/model/CourseCreateRequestDTO.kt
package kh.farrukh.sortmapperexample.course.model

import jakarta.validation.constraints.NotBlank
import jakarta.validation.constraints.Positive
import jakarta.validation.constraints.PositiveOrZero
import jakarta.validation.constraints.Size

data class CourseCreateRequestDTO(
    @field:[NotBlank Size(min = 2, max = 50)]
    val title: String,
    @field:[NotBlank Size(min = 2, max = 200)]
    val description: String,
    @field:Size(min = 2, max = 100)
    val slogan: String?,
    val teacherId: Long,
    @field:Positive
    val durationMonth: Int,
    @field:PositiveOrZero
    val price: Double
)
